### ISSUE TITLE
host on fly.io

### ISSUE THREAD

**Issue by IslamZaoui** [factoriotools/factorio-docker#494]
Is there any way to use this image on fly.io?

**Comment by fisher60** [comment]
Hello, this is the source code for a docker image. I did a quick search and it appears that fly.io is a host for deploying docker images. Following [their docs](https://fly.io/docs/getting-started/launch-demo/), you should be able to host this on fly.io.

I see the command should be:
`fly launch --image factoriotools/factorio:stable`. This would start the hosted image for this repo with the `stable` version tag. I am unsure what ports/permissions fly.io will allow, so hopefully you can expose the correct ports (sometimes this may cost extra or not be available if you do not have a VPS/server with the full port range available).

**Comment by miend** [comment]
The tl;dr is "yes".

Quick technical correction -- while fly.io processes docker images for app deployments, they don't actually use docker (or any other container engine) on the backend for running the final product. They extract the filesystem and run it on a tiny Firecracker VM, and you have wide permissions in there.

Ports can be exposed via `[[services]]` in the `fly.toml` [configuration file](https://fly.io/docs/reference/configuration/#the-services-sections) for your Fly app.

In this case you'd also need a [dedicated IPv4 address](https://fly.io/docs/networking/services/#dedicated-ipv4) because Fly's infra doesn't support UDP over shared IPv4s or IPv6.

**Comment by jamiemc1** [comment]
@IslamZaoui 
Did you manage to get this running?

Having a [crack](https://github.com/jamiemc1/factorio-server) on this too and am hitting some challenges.

**Comment by jamiemc1** [comment]
Spent another evening on it and got it working 😅

Reach out if you have any problems, or check out the linked repo.
